uui-components/src/pickers/hooks/usePicker.ts

```diff
@@ -0,0 +1,196 @@
+import { useCallback, useContext, useEffect, useRef } from 'react';
+import isEqual from 'lodash.isequal';
+import { DataRowOptions, DataSourceListProps, DataSourceState, IDataSourceView, PickerFooterProps, UuiContext } from '@epam/uui-core';
+import { PickerInputBaseProps } from '../PickerInputBase';
+import { applyValueToDataSourceState, dataSourceStateToValue } from '../bindingHelpers';
+import { PickerState } from './usePickerState';
+
+export function usePicker<TItem, TId>(
+    props: PickerInputBaseProps<TItem, TId>,
+    pickerState: PickerState,
+) {
+    const context = useContext(UuiContext);
+    const { showSelected, setShowSelected, dataSourceState, setDataSourceState } = pickerState;
+    const {
+        dataSource,
+        emptyValue,
+        value,
+        onValueChange,
+        getValueChangeAnalyticsEvent,
+        entityName,
+        entityPluralName,
+        selectionMode,
+        getSearchFields,
+        isFoldedByDefault,
+        sortBy,
+        cascadeSelection,
+    } = props;
+
+    const propsRef = useRef(props);
+    propsRef.current = props;
+
+    const pickerStateRef = useRef(pickerState);
+    pickerStateRef.current = pickerState;
+
+    const handleDataSourceValueChangeRef = useRef((newDataSourceState: DataSourceState) => {
```

Comment:
@jakobz, @AlekseyManetov, please, have a look at this piece of code. It is required to create this function via ref to keep the link because a view is created and retrieved via a link to this function later.